src/cargo/util/config.rs

```diff
@@ -21,6 +22,30 @@ use util::toml as cargo_toml;
 
 use self::ConfigValue as CV;
 
+#[derive(PartialEq, Eq, Hash)]
+struct ConfigFile {
+    path: Option<PathBuf>,
+}
+
+impl ConfigFile {
+    pub fn new(path: PathBuf) -> ConfigFile {
+        let canonical = match fs::canonicalize(path) {
```

Comment:
I believe this didn't previously call `fs::canonicalize`, so perhaps this could be left out?